src/main/java/uk/gov/ida/verifyserviceprovider/services/VerifyAndEidasKeyInfoCredentialResolver.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.ida.verifyserviceprovider.services;
+
+import com.google.common.collect.Lists;
+import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+import org.opensaml.security.credential.Credential;
+import org.opensaml.xmlsec.keyinfo.KeyInfoCredentialResolver;
+import org.opensaml.xmlsec.signature.support.impl.ExplicitKeySignatureTrustEngine;
+import uk.gov.ida.saml.metadata.EidasMetadataResolverRepository;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.LinkedList;
+import java.util.List;
+
+public class VerifyAndEidasKeyInfoCredentialResolver implements KeyInfoCredentialResolver {
+
+    private final ExplicitKeySignatureTrustEngine hubTrustEngine;
+    private final EidasMetadataResolverRepository eidasMetadataResolverRepository;
+
+    public VerifyAndEidasKeyInfoCredentialResolver(ExplicitKeySignatureTrustEngine hubTrustEngine, EidasMetadataResolverRepository eidasMetadataResolverRepository) {
+        this.hubTrustEngine = hubTrustEngine;
+        this.eidasMetadataResolverRepository = eidasMetadataResolverRepository;
+    }
+
+    @Nonnull
+    @Override
+    public Iterable<Credential> resolve(@Nullable CriteriaSet criteria) throws ResolverException {
+        List<Credential> credentialsFound = new LinkedList<>();
+        if (criteria == null) {
+            return credentialsFound;
+        }
+
+        for (String eidasResolverEntityId : eidasMetadataResolverRepository.getResolverEntityIds()) {
+            eidasMetadataResolverRepository.getSignatureTrustEngine(eidasResolverEntityId).ifPresent(engine -> {
+                try {
+                    KeyInfoCredentialResolver eidasCredentialResolver = engine.getKeyInfoResolver();
+                    credentialsFound.addAll(Lists.newLinkedList(eidasCredentialResolver.resolve(criteria)));
+                } catch (ResolverException e) {
+                    // NOP - could not resolve the criteria
```

Comment:
Use `} catch (ResolverException ignored) {` instead of comment

Reply:
👍